printer.go

```diff
@@ -136,6 +173,120 @@ type DriverInfo struct {
 	Attributes  uint32
 }
 
+// PrintJobInfo stores information about a print job.
+type PrintJobInfo struct {
+	JobID           uint32
+	UserMachineName string
+	UserName        string
+	DocumentName    string
+	DataType        string
+	Status          string
+	StatusCode      uint32
+	Priority        uint32
+	Position        uint32
+	TotalPages      uint32
+	PagesPrinted    uint32
+	Submitted       time.Time
+}
+
+// Jobs returns information about all print jobs on this printer
+func (p *Printer) Jobs() ([]PrintJobInfo, error) {
+	var bytesNeeded, jobsReturned uint32
+	buf := make([]byte, 1)
+	for {
+		err := EnumJobs(p.h, 0, 255, 1, &buf[0], uint32(len(buf)), &bytesNeeded, &jobsReturned)
+		if err == nil {
+			break
+		}
+		if err != syscall.ERROR_INSUFFICIENT_BUFFER {
+			return nil, err
+		}
+		if bytesNeeded <= uint32(len(buf)) {
+			return nil, err
+		}
+		buf = make([]byte, bytesNeeded)
+	}
+	if jobsReturned <= 0 {
+		return nil, nil
+	}
+	pj := make([]PrintJobInfo, 0, jobsReturned)
+	ji := (*[2048]JOB_INFO_1)(unsafe.Pointer(&buf[0]))[:jobsReturned]
+	for idx, j := range ji {
+		pj = append(pj, PrintJobInfo{
```

Comment:
Just a suggestion - feel free to ignore. But I think you can make this code clearer, if you create new variable here instead:
```
pjj := PrintJobInfo{
....
```
and move append on line 215 to the bottom of this for loop as
```
pj = append(pj, pjj)
```
Then you won't need to idx variable.

Maybe even use different variable names, for example:
```
pj := PrintJobInfo{
```
and
```
pjs = append(pjs, pj)
```